internal/core/platform.rs

```diff
@@ -36,6 +36,14 @@ pub trait Platform {
         Err(PlatformError::NoEventLoopProvider)
     }
 
+    /// Spins an event loop for a specified period of time.
```

Comment:
It's true that at least in winit many implementations of `run_on_demand()` (our `run_event_loop()`) basically call `pump_events()` (this `process_events()`). But it's not 100% the same, for example on x11 there's what looks like an extra `XSync()` call, and on macOS there's some different about stopping the NSApp after receiving `draw_rect()` (redraw request).

At this point I'd like to play it safe and not merge the two, to reduce the risk of regressions. Instead, I'll mark it as `#[doc(hidden)]` so that we have options to change it in the future. The same applies to the return value, maybe we'll find out that we need something in addition.